### ISSUE TITLE
Resource does not use updated value from another resource

### ISSUE THREAD

**Issue by johansnow** [hashicorp/terraform#18752]
I want to specifically ignore changes to what docker image is used for some aws ecs services, while falling back to Terraform-supplied values for new services or environments. To achieve this, I am

 1. using data "external" to fetch the data from aws
 2. "null_resource" triggers on this data to have it evaluated at planning stage
 3. data "template_file" with container definitions for the ecs task referring to the null_resources.triggers.value to use the fetched information

The problem is that if I run Terraform plan after a service starts using a new task def with another docker image, the following happens
 1. Through the null_resource trigger, the external data source correctly fetches the updated information from aws
 2. Terraform wants to update the null_resource with the newly fetched information
 3. Terraform wants to create a new task definition, but does not use the updated value supplied through the null_resource (the old value is used)

Let's say the initial docke image was foo/1 and task definition was foo-service:1. Through non-Terraform means, foo/2 is the docker image used by foo-service:2.

Running Terraform apply multiple times follow this pattern

 1. null-resource is updated to foo/2 (correct), and a new task def revision is created: foo-service:3, which uses the docker image foo/1 (incorrect)
 2. null-resource is updated to foo/1 (correct, because this is the docker image incorrectly put into service in the previous step). a new task def revision is created: foo-service:4, which uses docker image foo/2 (incorrect)
 3. null resource is updated to foo/2 (correct, because this is the docker image incorrectly put into service in the previous step). a new task def revision is created: foo-service:5, which uses docker image foo/1 (incorrect)
Because the container definition for the task definition is using outdated information, subsequent Terraform apply will alternate between these two images. This goes on until there is another external, and then the same alternating pattern starts with those two docker images

The Terraform script reduced to the minimum information I believe is relevant for the issue
```HIL
resource "aws_ecs_task_definition" "task_def" {
  count = "${length(var.services)}"

  container_definitions = "${element(data.template_file.container_definition.*.rendered, count.index)}"
}
data "template_file" "container_definition" {
  count = "${length(var.services)}"

  template = "${file("${path.module}/task_definition.json.tpl")}"

  vars {
    image                    = "${element(null_resource.read-service-image.*.triggers.image, count.index)}"
  }
}
data "external" "service_image" {
  count = "${length(var.services)}"
  program = ["/bin/bash", "${path.module}/get-service-image.sh"]
}
resource "null_resource" "read-service-image" {
  count = "${length(var.services)}"
  triggers {
    image = "${element(data.external.service_image.*.result.image, count.index)}"
  }
}
```

I asked a question on stack overflow (https://stackoverflow.com/questions/52045123/how-to-get-terraform-template-for-container-definitions-use-updated-value-from-n)  about how to get the above scenario to work, but after thinking about it more, I believe this is an actual issue, and that it is an issue with Terraform rather than the AWS Provider. Please let me know if this is incorrect so I may update the question on stack overflow or open an issue on the AWS Provider repo instead.

**Comment by johansnow** [comment]
I have found a way to achieve what I want without using the intermediate null_resource.

TLDR: The null_resource is no longer needed after upgrading to Terraform 0.11.8 and AWS Provider 1.33.0. My terraform script thus behaves the way I intended it to, after version upgrades followed by removing the null_resource.

Background: During my first attempt, I did use the null_resource. However, when running Terraform 0.11.7 and AWS Provider 1.32.0 the planning stage did not include the information retrieved by my data.external. I introduced the null_resource to make sure that part was evaluated during planning, which lead to the problems described by this issue.
So while my particular use case no longer needs it, it would seem that the underlying issue still remains.
